docs/libraries/typespec-client-generator-core/rules/CONTRIBUTING.md

```diff
@@ -0,0 +1,36 @@
+# Contributing Linter Rules to the @azure-tools/typespec-client-generator-core Package
+
+For information about linter rules in typespec in general, view [here][generic-linter]
+
+In the `@azure-tools/typespec-client-generator-core` library, we have two main types of rules:
+
+1. Generic rule that applies to all emitted languages
+2. Specific rule that applies to a subset of all language: it can even apply to just one language
+
+The process for adding a rule starts off the same for both: for language-specific rules, there is an extra step
+
+1. Write the rule in `typespec-azure/packages/typespec-client-generator-core/src/rules/[rule-name].rule.ts
+2. Add reference to the rule in the `rules` array in [`typespec-azure/packages/typespec-client-generator-core/src/linter.ts`][tcgc-linter]
+   - This will automatically add it to a ruleset called `:all` for `@azure-tools/typespec-client-generator-core`
+3. Add the rule to the enable list for [`data-plane.ts`][data-plane-ruleset] and/or [`resource-manager.ts`][resource-manager-ruleset] in the [rulesets][rulesets] package. You can set `enable` to `false` here, if you want to delay enabling
```

Comment:
Do we really want to have different rules for data plane and mgmt plane?  I know we are less strict with mgmt plane, but aren't the recommendations the same for both?

Reply:
That's the current setup of `azure-core-rulesets`, where you explicitly list rules for the [data plane](https://github.com/Azure/typespec-azure/blob/main/packages/typespec-azure-rulesets/src/rulesets/data-plane.ts) and [mgmt](https://github.com/Azure/typespec-azure/blob/main/packages/typespec-azure-rulesets/src/rulesets/resource-manager.ts) rulesets

Reply:
there is some rules that management plane specific and some that shouldn't apply to mamangement plane. But the majority should apply to both

Reply:
Do we have a single place to add a rule that applies to both or do we need to add the rule explicitly to both lists?